apps/rush-lib/src/logic/deploy/DeployManager.ts

```diff
@@ -479,7 +498,19 @@ export class DeployManager {
     });
   }
 
-  private _prepareDeployment(deployState: IDeployState): void {
+  private async _makeBinLinks(deployState: IDeployState) {
+    for (const [, folderInfo] of deployState.folderInfosByPath) {
+      const deployedPath = this._remapPathForDeployMetadata(folderInfo.folderPath, deployState);
+      const projectFolder: string = path.join(deployedPath, 'node_modules');
+      const projectBinFolder: string = path.join(deployedPath, 'node_modules', '.bin');
```

Comment:
These need to be absolute paths, like this:

```ts
      const projectFolder: string = path.join(deployState.targetRootFolder, deployedPath, 'node_modules');
      const projectBinFolder: string = path.join(deployState.targetRootFolder, deployedPath, 'node_modules', '.bin');
```

Otherwise `pnpmLinkBins()` (frustratingly) silently does nothing at all.

BTW there's some tips for debugging Rush [on this page](https://rushjs.io/pages/contributing/).

Reply:
With the above fix, the binaries are created and work correctly.